src/Common/ThreadStatus.cpp

```diff
@@ -172,9 +172,19 @@ ThreadGroupPtr ThreadStatus::getThreadGroup() const
     return thread_group;
 }
 
+void ThreadStatus::setQueryId(std::string && new_query_id)
+{
+    query_id = std::move(new_query_id);
```

Comment:
It should be empty before, right? Worth adding `chassert(query_id.empty())` then